File: abc_spec/code_samples/Java/instruments@{id}/patch.java
// For more information please refer to https://github.com/checkout/checkout-sdk-java
import com.checkout.CheckoutApiException;
import com.checkout.CheckoutArgumentException;
import com.checkout.CheckoutAuthorizationException;
import com.checkout.CheckoutSdk;
import com.checkout.Environment;
import com.checkout.common.Address;
import com.checkout.common.CountryCode;
import com.checkout.common.Phone;
import com.checkout.instruments.previous.InstrumentAccountHolder;
import com.checkout.instruments.previous.UpdateInstrumentRequest;
import com.checkout.instruments.previous.UpdateInstrumentResponse;
import com.checkout.previous.CheckoutApi;

CheckoutApi api = CheckoutSdk
    .builder()
    .previous()
    .staticKeys()
    .secretKey("secret_key")
    .environment(Environment.SANDBOX) // or Environment.PRODUCTION
    .build();

UpdateInstrumentRequest updateRequest = UpdateInstrumentRequest.builder()
    .name("New name")
    .expiryMonth(10)
    .expiryYear(2025)
    .accountHolder(InstrumentAccountHolder.builder()
        .billingAddress(Address.builder()
            .addressLine1("Checkout")
            .addressLine2("90 Tottenham Court Road")
            .city("London")
            .state("London")
            .zip("W1T 4TJ")
            .country(CountryCode.GB)
            .build())
        .phone(Phone.builder().countryCode("1").number("415 555 2671").build())
        .build())
    .customer(UpdateInstrumentRequest.Customer.builder()
        .id("customer_id")
        .isDefault(true)
        .build())
    .build();

try {
    UpdateInstrumentResponse response = api.instrumentsClient().update("instrument_id", updateRequest).get();
} catch (CheckoutApiException e) {
    // API error
    String requestId = e.getRequestId();
    int statusCode = e.getHttpStatusCode();
    Map<String, Object> errorDetails = e.getErrorDetails();
} catch (CheckoutArgumentException e) {
    // Bad arguments
} catch (CheckoutAuthorizationException e) {
    // Invalid authorization
}

File: abc_spec/code_samples/Python/instruments@{id}/patch.py
# For more information please refer to https://github.com/checkout/checkout-sdk-python
import checkout_sdk
from checkout_sdk.checkout_sdk import CheckoutSdk
from checkout_sdk.common.common import Phone, Address
from checkout_sdk.common.enums import Country
from checkout_sdk.environment import Environment
from checkout_sdk.exception import CheckoutApiException, CheckoutArgumentException, CheckoutAuthorizationException
from checkout_sdk.instruments.instruments_previous import UpdateInstrumentCustomer, InstrumentAccountHolder, UpdateInstrumentRequest

api = CheckoutSdk.builder() \\
    .previous() \\
    .secret_key('secret_key') \\
    .environment(Environment.sandbox()) \\
    .build()
    # or Environment.production()

phone = Phone()
phone.country_code = '44'
phone.number = '4155552671'

address = Address()
address.address_line1 = 'CheckoutSdk.com'
address.address_line2 = '90 Tottenham Court Road'
address.city = 'London'
address.state = 'London'
address.zip = 'W1T 4TJ'
address.country = Country.GB

update_instrument_customer = UpdateInstrumentCustomer()
update_instrument_customer.id = 'id'
update_instrument_customer.default = True

account_holder = InstrumentAccountHolder()
account_holder.billing_address = address
account_holder.phone = phone

update_instrument_request = UpdateInstrumentRequest()
update_instrument_request.name = 'New Name'
update_instrument_request.expiry_year = 2027
update_instrument_request.expiry_month = 12
update_instrument_request.customer = update_instrument_customer
update_instrument_request.account_holder = account_holder

try:
    response = api.instruments.update('instrument_id', update_instrument_request)
except CheckoutApiException as err:
    # API error
    request_id = err.request_id
    status_code = err.http_status_code
    error_details = err.error_details
    http_response = err.http_response
except CheckoutArgumentException as err:
    # Bad arguments

except CheckoutAuthorizationException as err:
    # Invalid authorization
